crates/tauri-bundler/src/bundle/windows/util.rs

```diff
@@ -69,18 +69,58 @@ pub fn download_webview2_offline_installer(base_path: &Path, arch: &str) -> crat
   Ok(file_path)
 }
 
-fn create_agent_and_url(url: &str) -> crate::Result<(ureq::Agent, String)> {
-  match std::env::var("TAURI_BUNDLER_TOOLS_GITHUB_MIRROR") {
-    Ok(cdn) if url.starts_with("https://github.com/") => {
+fn generate_mirror_url_from_template(github_url: &str) -> Option<String> {
+  std::env::var("TAURI_BUNDLER_TOOLS_GITHUB_MIRROR_TEMPLATE")
+    .ok()
+    .and_then(|template| {
+      let re =
+        Regex::new(r"https://github.com/([^/]+)/([^/]+)/releases/download/([^/]+)/(.*)").unwrap();
+      re.captures(github_url).map(|caps| {
+        template
+          .replace("<owner>", &caps[1])
+          .replace("<repo>", &caps[2])
+          .replace("<version>", &caps[3])
+          .replace("<asset>", &caps[4])
+      })
+    })
+}
+
+fn generate_mirror_url_from_base(github_url: &str) -> crate::Result<Option<String>> {
+  std::env::var("TAURI_BUNDLER_TOOLS_GITHUB_MIRROR")
+    .ok()
+    .map(|cdn| {
       let mut parsed_cdn = Url::parse(&cdn)?;
-      parsed_cdn.set_path(url);
-      Ok((AgentBuilder::new().build(), parsed_cdn.into()))
-    }
-    _ => Ok((
-      AgentBuilder::new().try_proxy_from_env(true).build(),
-      url.to_owned(),
-    )),
+      parsed_cdn.set_path(github_url);
+      Ok(parsed_cdn.into())
+    })
+    .transpose()
+}
+
+fn generate_alternative_url(url: &str) -> crate::Result<Option<(ureq::Agent, String)>> {
+  if !url.starts_with("https://github.com/") {
+    return Ok(None);
   }
+
+  let mirror_url = generate_mirror_url_from_template(url);
+  let cdn_url = generate_mirror_url_from_base(url)?;
+
+  Ok(
+    mirror_url
+      .or(cdn_url)
+      .map(|alternative_url| (ureq::AgentBuilder::new().build(), alternative_url)),
+  )
```

Comment:
This can be simplified into

```suggestion
  generate_mirror_url_from_template(url)
    .or_else(|_| generate_mirror_url_from_base(url))
    .map(|alt_url| {
      alt_url.map(|alt_url| (ureq::AgentBuilder::new().build(), alt_url))
    })
```

also notice usage of `or_else` instead of `or` which will lazily evaluate `generate_mirror_url_from_base` only if needed.